RHIDP-1925 bulk import from GitHub (#616)
redhat-developer/red-hat-developers-documentation-rhdh: modules/importing-repositories/procedure-understanding-bulk-import-audit-logs.adoc

= Bulk Import Audit Logs

> The Bulk Import Backend plugin adds the events below to the Developer Hub audit logs.

Reviewer: ```suggestion
The Bulk Import backend plugin adds the following events to the {product-short} audit logs.
```

See link:{linkgettingstartedguide}#assembly-audit-log[Audit Logs in {product}] for more information on how to configure and view such audit logs.

*Bulk Import Events*:

`BulkImportUnknownEndpoint`::
Tracks requests to unknown endpoints.

`BulkImportPing`::
Tracks `GET` requests to the `/ping` endpoint, which allows us to make sure the bulk import backend is up and running.

`BulkImportFindAllOrganizations`::

> Tracks `GET` requests to the /organizations endpoint, which returns the list of organizations accessible from all configured GitHub Integrations.

Reviewer: ```suggestion
Tracks `GET` requests to the `/organizations` endpoint, which returns the list of organizations accessible from all configured GitHub Integrations.
```

`BulkImportFindRepositoriesByOrganization`::
Tracks `GET` requests to the `/organizations/:orgName/repositories` endpoint, which returns the list of repositories for the specified organization (accessible from any of the configured GitHub Integrations).

`BulkImportFindAllRepositories`::
Tracks GET requests to the `/repositories` endpoint, which returns the list of repositories accessible from all configured GitHub Integrations.

`BulkImportFindAllImports`::
Tracks `GET` requests to the `/imports` endpoint, which returns the list of existing import jobs along with their statuses.

`BulkImportCreateImportJobs`::

> Tracks `POST` requests to the `/imports` endpoint, which allows to submit requests to bulk-import one or many repositories into the Developer Hub Catalog, by eventually creating import Pull Requests in the target repositories.

Reviewer: ```suggestion
Tracks `POST` requests to the `/imports` endpoint, which allows to submit requests to bulk-import one or many repositories into the {product-short} catalog, by eventually creating import pull requests in the target repositories.
```

`BulkImportFindImportStatusByRepo`::
Tracks `GET` requests to the `/import/by-repo` endpoint, which fetches details about the import job for the specified repository.

`BulkImportDeleteImportByRepo`::

> Tracks `DELETE` requests to the `/import/by-repo` endpoint, which deletes any existing import job for the specified repository, by closing any open import Pull Request that could have been created.

Reviewer: ```suggestion
Tracks `DELETE` requests to the `/import/by-repo` endpoint, which deletes any existing import job for the specified repository, by closing any open import pull request that could have been created.
```

> .Example

Reviewer: ```suggestion
.Example bulk import audit logs
```

[id="understanding-bulk-import-audit-logs"]
= Understanding the Bulk Import audit Logs

The Bulk Import backend plugin adds the following events to the {product-short} audit logs.

> See link:{linkgettingstartedguide}#assembly-audit-log[Audit Logs in {product}] for more information on how to configure and view such audit logs.

Reviewer: ```suggestion
See link:{linkgettingstartedguide}#assembly-audit-log[Audit Logs in {product}] for more information on how to configure and view audit logs.
```
